hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -587,7 +587,7 @@ void addToCompactionLogTable(CompactionLogEntry compactionLogEntry) {
    * Check if there is any in_progress tarball creation request and wait till
    * all tarball creation finish, and it gets notified.
    */
-  private void waitForTarballCreation() {
+  private synchronized void waitForTarballCreation() {
```

Comment:
`tarballRequestCount` is an `AtomicInteger` so the incrementing and decrementing it would be atomic.

`waitForTarballCreation` is synchronized because wait/notify are supposed to be allowed on the same object's lock otherwise it throws `IllegalMonitorStateException`. Doc: [wait](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#wait--)/[notifyAll](https://docs.oracle.com/javase/8/docs/api/java/lang/Object.html#notifyAll--).

More in discussion on jira: HDDS-9039 why we are adding the test this way.

Reply:
What I didn't understand is getCheckpoint() method is not taking a synchronized lock. So the question if waitForTarballCreation() passes. https://github.com/apache/ozone/blob/2c0580dbb2edeb94ec263aa44fd1e22cff838dcc/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java#L248 can still run. Thus isn't there a race condition here? So while compaction is running we still end up starting a tarballing for the checkpoint.

Reply:
I think you are right if we are appending compaction log entries to a txt file but we don't use that anymore and add entries to compactionLog columnFamily.

I gave it more thought and I think this locking is not needed anymore because we use RocksDB column family for compaction now. So compaction entry will be either present in the table or not in the snapshot of the ActiveFS depending on the order or compaction entry append and checkpoint creation. Hence we can simply remove this locking code and just rely on RocksDB.

Reply:
yeah i agree we don't need the locks since we are writing it into a rocksdb table. This is fine as long as we are using rocksdb checkpoints to take a checkpoint of the rocksdb.